core/husarnet/api_server/dashboard_api_proxy.cpp

```diff
@@ -0,0 +1,38 @@
+// Copyright (c) 2024 Husarnet sp. z o.o.
+// Authors: listed in project_root/README.md
+// License: specified in project_root/LICENSE.txt
+#include "husarnet/api_server/dashboard_api_proxy.h"
+
+#include "husarnet/logging.h"
+
+bool DashboardApiProxy::isValid()
+{
+  return !apiAddress.empty();
+}
+
+void DashboardApiProxy::signAndForward(
```

Comment:
Normally I'd say that doing HTTP requests should be a feature `Port::` should expose but… this is so concise that it's better to leave it here rather than making it 100% bulletproof for all platforms (especially embedded ones)